rotkehlchen/exchanges/kraken.py

```diff
@@ -800,28 +725,25 @@ def process_kraken_trades(
                     )
                     continue
 
-                rate = Price(abs(receive_event.amount / spend_event.amount))
-                trade = Trade(
-                    timestamp=ts_ms_to_sec(a1.timestamp),
+                swap_events.extend(create_swap_events(
+                    timestamp=a1.timestamp,
                     location=Location.KRAKEN,
-                    base_asset=receive_event.asset,
-                    quote_asset=spend_event.asset,
-                    trade_type=TradeType.BUY,
-                    amount=AssetAmount(receive_event.amount),
-                    rate=rate,
-                    fee=None,
-                    fee_currency=None,
-                    link='adjustment' + a1.event_identifier + a2.event_identifier,
-                )
-                trades.append(trade)
+                    spend_asset=spend_event.asset,
+                    spend_amount=spend_event.amount,
+                    receive_asset=receive_event.asset,
+                    receive_amount=receive_event.amount,
+                    fee_asset=A_USD,  # whatever
```

Comment:
I don't understaht this "whatever" thing. If it can be whatever, then it should be `None`. Otherwise the logic makes no sense. Please fix this in the definition of `create_swap_events`. if should check if `fee_asset` is None`. And both `fee_asset` and `fee_amount` should be None/Noneable.